plugins/vpp/l2plugin/vppcalls/interface_vppcalls.go

```diff
@@ -18,76 +18,85 @@ import (
 	"fmt"
 	"time"
 
+	"github.com/go-errors/errors"
+
 	"github.com/ligato/cn-infra/logging"
 	l2ba "github.com/ligato/vpp-agent/plugins/vpp/binapi/l2"
 	"github.com/ligato/vpp-agent/plugins/vpp/ifplugin/ifaceidx"
 	"github.com/ligato/vpp-agent/plugins/vpp/model/l2"
 )
 
-// SetInterfacesToBridgeDomain implements bridge domain handler.
-func (handler *BridgeDomainVppHandler) SetInterfacesToBridgeDomain(bdName string, bdIdx uint32, bdIfs []*l2.BridgeDomains_BridgeDomain_Interfaces,
-	swIfIndices ifaceidx.SwIfIndex) (ifs []string, wasErr error) {
+// SetInterfaceToBridgeDomain implements bridge domain handler. Returns an interface configured to the BD.
+func (h *BridgeDomainVppHandler) SetInterfaceToBridgeDomain(bdName string, bdIdx uint32, bdIf *l2.BridgeDomains_BridgeDomain_Interfaces,
+	swIfIndices ifaceidx.SwIfIndex) (string, error) {
 	defer func(t time.Time) {
-		handler.stopwatch.TimeLog(l2ba.SwInterfaceSetL2Bridge{}).LogTimeEntry(time.Since(t))
+		h.stopwatch.TimeLog(l2ba.SwInterfaceSetL2Bridge{}).LogTimeEntry(time.Since(t))
 	}(time.Now())
 
-	if len(bdIfs) == 0 {
-		handler.log.Debugf("Bridge domain %v has no new interface to set", bdName)
-		return nil, nil
+	// Verify that interface exists, otherwise skip it.
+	ifIdx, _, found := swIfIndices.LookupIdx(bdIf.Name)
+	if !found {
+		h.log.Debugf("Required bridge domain %s interface %s not found", bdName, bdIf.Name)
+		return "", nil
+	}
+	if err := h.addDelInterfaceToBridgeDomain(bdName, bdIdx, bdIf, ifIdx, true); err != nil {
+		return "", err
 	}
+	h.log.WithFields(logging.Fields{"Interface": bdIf.Name, "BD": bdName}).Debug("Interface set to bridge domain")
 
+	return bdIf.Name, nil
+}
+
+// SetInterfacesToBridgeDomain implements bridge domain handler. Returns a list of interfaces configured to the BD.
+func (h *BridgeDomainVppHandler) SetInterfacesToBridgeDomain(bdName string, bdIdx uint32, bdIfs []*l2.BridgeDomains_BridgeDomain_Interfaces,
```

Comment:
Seems like we could get rid of this. It's not used anywhere now (if I didn't miss it) and it doesn't really provide any value which could not be done in the higher layer. Let's keep the handlers smaller if possible.

Reply:
still used [here](https://github.com/ligato/vpp-agent/blob/master/plugins/vpp/l2plugin/bd_config.go#L143) and [here](https://github.com/ligato/vpp-agent/blob/master/plugins/vpp/l2plugin/bd_config.go#L218) (original code)

of course, it can be replaced by for loops and method setting single interface I added, but I think it's worth to keep it here, since it would add some code redundancy